server/workflowNodes.ts

```
@@ -357,20 +367,47 @@ registerNodeExecutor("action_db", async (ctx) => {
 
     switch (operation) {
       case "insert":
-        // Simple insert (in production, use proper Drizzle queries)
-        result = await db.execute(`INSERT INTO ${table} SET ?`);
+        if (!data || typeof data !== "object" || Array.isArray(data)) {
+          throw new Error("Data must be an object for insert");
+        }
+        const keys = Object.keys(data);
```

Biome (2.4.10)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)